app/models/api/hses.rb

```diff
@@ -1,25 +1,53 @@
+require "api_request"
 require "fake_api_response_wrapper"
 require "fake_issues"
 
-class Api::Hses
-  class Issue
-    include FakeApiResponseWrapper
-    attr_accessor :id
+module Api::Hses
+end
+
+class Api::Hses::Issue < ApiRequest
+  include FakeApiResponseWrapper
+  attr_accessor :id
+
+  def initialize(id:)
+    @id = id
+  end
+
+  # TODO update this with real data once we have /issue endpoint
+  def request
+    details_wrapper.merge(
+      data: FakeIssues.instance.json[:data].sample
+    )
+  end
+
+  private
+
+  def host
+    Rails.configuration.x.hses.api_base
+  end
+end
+
+class Api::Hses::Issues < ApiRequest
+  def initialize(user:)
+    @username = user["uid"]
+  end
 
-    def initialize(id:)
-      @id = id
-    end
+  def request
+    res = response
```

Comment:
I don't like this line very much, so I'm happy to take some suggestions on what to do about this.  I think what I would like, ideally, is that `response` isn't just a hash, but is an instance of an ApiResponse class with things like `response.code`, `response.data`, `response.current_page` and that type of thing.  For now, this is enough to get us up and running, I think.

Reply:
since we're creating a new Api::Hses::Issues instance each time we call this, can we memoize `response` in ApiRequest?